scripts/globals/combat/physical_utilities.lua

```diff
@@ -80,6 +80,102 @@ local shieldSizeToBlockRateTable =
+    end
+
+    -- Consume Mana
+    if attacker:isPC() then
+        bonusBasePhysicalDamage = bonusBasePhysicalDamage + xi.combat.effect.handleConsumeMana(attacker)
+    end
+
+    if isH2H then
+        local naturalH2hDamage = math.floor(attacker:getSkillLevel(xi.skill.HAND_TO_HAND) * 0.11) + 3
+
+        if physicalAttackType == xi.physicalAttackType.KICK then
+            local kickDamage = naturalH2hDamage + attacker:getMod(xi.mod.KICK_DMG)
+            damage = math.floor(damageRatio * (kickDamage + bonusBasePhysicalDamage + xi.combat.physical.calculateMeleeStatFactor(attacker, victim)))
+        else
+            damage = math.floor(damageRatio * (baseDamage + naturalH2hDamage + bonusBasePhysicalDamage + xi.combat.physical.calculateMeleeStatFactor(attacker, victim)))
+        end
+    elseif slot == xi.slot.MAIN then
+        damage = math.floor(damageRatio * (baseDamage + bonusBasePhysicalDamage + xi.combat.physical.calculateMeleeStatFactor(attacker, victim)))
+    elseif slot == xi.slot.AMMO then
+        damage = math.floor(damageRatio * (baseDamage + xi.combat.physical.calculateMeleeStatFactor(attacker, victim)))
+    end
+
+    -- Scarlet Delirium
+    damage = math.floor(damage * xi.combat.damage.scarletDeliriumMultiplier(attacker))
+
+    -- Double Attack and Triple Attack
+    if physicalAttackType == xi.physicalAttackType.DOUBLE and attacker:isPC() then
+        damage = math.floor(damage * ((100 + attacker:getMod(xi.mod.DOUBLE_ATTACK_DMG)) / 100))
+    elseif physicalAttackType == xi.physicalAttackType.TRIPLE and attacker:isPC() then
+        damage = math.floor(damage * ((100 + attacker:getMod(xi.mod.TRIPLE_ATTACK_DMG)) / 100))
+    end
+
+    -- Soul Eater
+    if attacker:isPC() then
+        damage = xi.combat.effect.handleSoulEater(attacker, damage)
+    end
+
+    -- Damage multipliers
+    damage = attacker:addDamageFromMultipliers(damage, physicalAttackType, slot, isFirstSwing)
+
+    -- Sneak Attack Augment
+    if
+        attacker:getMod(xi.mod.AUGMENTS_SA) > 0 and
+        isSneakAttack and
+        attacker:hasStatusEffect(xi.effect.SNEAK_ATTACK)
+    then
+        damage = damage + math.floor(damage * ((100 + attacker:getMod(xi.mod.AUGMENTS_SA)) / 100))
+    end
+
+    -- Trick Attack Augment
+    if
+        attacker:getMod(xi.mod.AUGMENTS_TA) > 0 and
+        isTrickAttack and
+        attacker:hasStatusEffect(xi.effect.TRICK_ATTACK)
+    then
+        damage = damage + math.floor(damage * ((100 + attacker:getMod(xi.mod.AUGMENTS_TA)) / 100))
+    end
+
+    --- Low level mobs can get negative fSTR so low they crater their (base weapon damage + fstr) to below 0.
+    --- Absorption isn't possible at this point in the calculation, so zero it.
+    if damage < 0 then
+        damage = 0
+    end
+
+    -- Apply Restraint Weaponskill Damage
+    if isFirstSwing then
+        xi.combat.effect.handleRestraint(attacker)
+    end
+
+    return damage
```

Comment:
Can you add a TODO before the return

```suggestion
    -- TODO: add charutils::TrySkillUP call
    return damage
```

Reply:
Reread the PR, you have the skillup left over in C++, but i'd still eventually like it in lua (this is more of a reminder comment)